tests/unit/s2n_build_test.c

```diff
@@ -0,0 +1,149 @@
+/*
+* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+#define _GNU_SOURCE
+#include <ctype.h>
+#include <openssl/crypto.h>
+#include <openssl/opensslv.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "s2n_test.h"
+
+struct libcrypto_info {
+    char *name;
+    int major_version;
+    int minor_version;
+    int patch_version;
+};
+
+int number_from_version(char *version)
```

Comment:
Our library prefers output arguments rather than returning the output. So this function ideally would be written like:
```suggestion
int str_to_int(char *input, int *output)
```
This is probably not super important in testing, but it's good practice to see how we write code for our library. This way we can separate return codes from output, like atoi() can't 🙄.